src/services/TestDataService.ts

```diff
@@ -0,0 +1,1234 @@
+     *
+     * @param key - The key of the salutation. Default is "mr".
+     */
+    async getSalutation(key = 'mr') {
+        const response = await this.AdminApiClient.post('search/salutation', {
+            data: {
+                limit: 1,
+                filter: [{
+                    type: 'equals',
+                    field: 'salutationKey',
+                    value: key,
+                }],
+            },
+        });
+
+        const { data: result } = await response.json();
+
+        return result[0];
+    }
+
+    /**
+     * Retrieves the state machine for order states.
+     */
+    async getOrderStateMachine(): Promise<StateMachine> {
+        return await this.getStateMachine('order.state');
+    }
+
+    /**
+     * Retrieves the state machine for delivery states.
+     */
+    async getDeliveryStateMachine(): Promise<StateMachine> {
+        return await this.getStateMachine('order_delivery.state');
+    }
+
+    /**
+     * Retrieves the state machine for transaction states.
+     */
+    async getTransactionStateMachine(): Promise<StateMachine> {
+        return await this.getStateMachine('order_transaction.state');
+    }
+
+    /**
+     * Retrieves a state machine by its name.
+     *
+     * @param name - The name of the state machine.
+     */
+    async getStateMachine(name: string): Promise<StateMachine> {
+        const response = await this.AdminApiClient.post('search/state-machine', {
+            data: {
+                limit: 1,
+                filter: [{
+                    type: 'equals',
+                    field: 'technicalName',
+                    value: name,
+                }],
+            },
+        });
+
+        const { data: result } = (await response.json()) as { data: StateMachine[] };
+
+        return result[0];
+    }
+
+    /**
+     * Retrieves the state of a state machine.
+     *
+     * @param stateMachineId - The uuid of the state machine.
+     */
+    async getStateMachineState(stateMachineId: string): Promise<StateMachineState> {
+        const response = await this.AdminApiClient.post('search/state-machine-state', {
+            data: {
+                limit: 1,
+                filter: [{
+                    type: 'equals',
+                    field: 'stateMachineId',
+                    value: stateMachineId,
+                }],
+            },
+        });
+
+        const { data: result } = (await response.json()) as { data: StateMachineState[] };
+
+        return result[0];
+    }
+
+    /**
+     * Set the configuration of automated data clean up.
+     * If set to "true" the data service will delete all entities created by it.
+     *
+     * @param shouldCleanUp - The config setting for the automated data clean up. Default is "true".
+     */
+    setCleanUp(shouldCleanUp = true) {
+        this.shouldCleanUp = shouldCleanUp;
+    }
+
+    /**
+     * Will delete all entities created by the data service via sync API.
+     */
+    async cleanUp() {
```

Comment:
Will be interesting to see, if this works for all entities. On principle it should work, if you delete all entities that were created in reverse order. I think some entities like orders, customers, sales channel might need custom logic.

Reply:
I had it as separate delete calls first and then changed it to one single sync API call. My hope was that within the sync call, it would resolve associations correctly. It seemed to work already for orders and customers.